logviewer/logviewer-lib.pl

```diff
@@ -27,33 +27,135 @@ sub can_edit_log
 return 0;
 }
 
-# 
-# Returns standard 
+# get_journal_since
+# Returns a list of journalctl commands to get logs since various times,
+# which should correspond with language strings journal_since0,
+# journal_since1, journal_since2, etc.
+sub get_journal_since
```

Comment:
Why not just pass in the time in seconds or something, and have the caller determine the time range?

Reply:
Because this is also used as a template to validate user input.

Reply:
I don't understand?

Reply:
Look, `get_journal_since` is the source of truth for **Since** select options, and what it shows in the UI. It is also used to validate user input (i.e. passed with `$in{'since'}`) to make sure nothing arbitrary gets to the command execution..

Reply:
Oh I see now ... this is just returning the options.  But are we displaying these exact strings to the user? What about localization?

Reply:
Ah, I see. I explained it in the comment above:

```perl
# get_journal_since
# Returns a list of journalctl commands to get logs since various times,
# which should correspond with language strings journal_since0,
# journal_since1, journal_since2, etc.
```

Reply:
But there is a better way of doing it https://github.com/webmin/webmin/commit/a420c7142faf66313dcf4e2686cf3969a3bdb820.

Reply:
Nice, I like that more

Reply:
> Nice, I like that more

Yeah, it's better. A little bit more complex though.